Compile fails on Linux without libbluray when using -DENABLE_BLURAY=OFF. Bisected to recent commit.

<!--- Please fill out this template to the best of your ability. You can always edit this issue once you have created it. -->
<!--- Read the following link before you create a new problem report: https://kodi.wiki/view/HOW-TO:Submit_a_bug_report  -->
## Bug report
### Describe the bug
Here is a clear and concise description of what the problem is:
<!--- Provide a more detailed introduction to the issue itself, and why you consider it to be a bug -->
<!--- A bug report that is not clear will be closed -->
<!--- Put your text below this line -->
Kodi git master fails to compile without libbluray installed even when using -DENABLE_BLURAY=OFF. After bisecting, the offending commit was by 78andyp on July 2, 2025: 6451b7937d811203597f902add034710a74958ab
"Fix retrieval of basepath."


## Expected Behavior
Here is a clear and concise description of what was expected to happen:
<!--- Tell us what should happen -->
<!--- Put your text below this line -->
Successfully compiling without libbluray.


## Actual Behavior
<!--- Tell us what happens instead -->
<!--- Put your text below this line -->
Fails to compile without libbluray.


## Possible Fix
<!--- Not obligatory, but suggest a fix or reason for the bug -->
<!--- Put your text below this line -->
Wrap the `#include filesystem/BlurayDirectory.h` in URIUtils.cpp with an `#ifdef HAVE_LIBBLURAY`.


### To Reproduce
Steps to reproduce the behavior:
<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code to reproduce, if relevant -->
<!--- Put your text below this line -->
1. If installed, temporarily uninstall libbluray-dev
2. Clone master 
3. Configure with -DENABLE_BLURAY=OFF and try compiling


### Debuglog
<!--- Put your text below this line -->
<!--- A debuglog is always mandatory when creating an issue. Provide one! -->
The debuglog can be found here:
There is no debug log because compile fails with:

In file included from /test/kodi-r68700.git+28fb9320ad-bisect/xbmc/utils/URIUtils.cpp:16:
/test/kodi-r68700.git+28fb9320ad-bisect/xbmc/filesystem/BlurayDirectory.h:18:10: fatal error: libbluray/bluray.h: No such file or directory
   18 | #include <libbluray/bluray.h>
      |          ^~~~~~~~~~~~~~~~~~~~
compilation terminated.



### Screenshots 
Here are some links or screenshots to help explain the problem:
<!--- Put your text below this line -->


## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:
<!--- How has this bug affected you? What were you trying to accomplish? -->
<!--- Put your text below this line -->


### Your Environment
Used Operating system:
<!--- Include as many relevant details about the environment you experienced the bug in -->
<!--- Put your text below this line. Checkboxes can easily be ticked once issue is created -->
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] macOS
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Debian Linux (sid)
 - Kodi version: Current git master (kodi-r68700.git+28fb9320ad)



<!--- End of this issue -->
*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

--- a/xbmc/utils/URIUtils.cpp
+++ b/xbmc/utils/URIUtils.cpp
@@ -13,7 +13,9 @@
 #include "ServiceBroker.h"
 #include "StringUtils.h"
 #include "URL.h"
+#ifdef HAVE_LIBBLURAY
 #include "filesystem/BlurayDirectory.h"
+#endif
 #include "filesystem/MultiPathDirectory.h"
 #include "filesystem/SpecialProtocol.h"
 #include "filesystem/StackDirectory.h"

